formver.compiler-plugin/core/src/org/jetbrains/kotlin/formver/core/embeddings/callables/FunctionEmbedding.kt

```diff
@@ -25,3 +31,16 @@ class UserFunctionEmbedding(private val callable: RichCallableEmbedding) : Funct
     override val viperMethod: Method?
         get() = body?.toViperMethod(callable) ?: callable.toViperMethodHeader()
 }
+
+
+/**
+ * An embedding of a user-defined pure function
+ */
+class PureUserFunctionEmbedding(private val callable: RichCallableEmbedding) : PureFunctionEmbedding,
+        CallableEmbedding by callable {
+
+    var body: PureFunctionBodyEmbedding? = null
+
+    override val viperFunction: Function?
+        get() = body?.toViperFunction(callable) ?: callable.toViperFunctionHeader()
```

Comment:
I think this is overly permissive - the idea of having a nullable bodyembedding for methods is that we only want to verify some methods.  But at the moment we want to translate all functions to ones with bodies, so that's quite a different story.